src/codegen/error.rs

```diff
@@ -0,0 +1,36 @@
+use std::error;
+use std::fmt;
+
+/// Errors that can occur during code generation.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Error {
```

Comment:
It is namespaced, however. I hope that it can become the error type for all fallible codegen operations.